tools/windows-node-installer/test/e2e/aws_test.go

```diff
@@ -454,3 +464,49 @@ func testInstallerJsonFileIsDeleted(t *testing.T) {
 	_, err := resource.ReadInstallerInfo(artifactDir)
 	assert.Error(t, err, "error deleting windows-node-installer.json file")
 }
+
+// mockCloudProviderFactory mocks the functionality of CloudProviderFactory. It uses OpenShift Client created by using
+// the service account in the test cluster. This function is required, as we cannot use inClusterConfig() directly in CI.
+// CI pod runs on 3.11 cluster, hence inClusterConfig() results configuration defined by 3.11 cluster. To mock the behaviour of
+// inClusterConfig we create client using configuration generated by a Service account with custom rbac rules. This client
+// is used to get the required cloud provider.
+func mockCloudProviderFactory(kubeconfigPath, credentialPath, credentialAccountID, resourceTrackerDir,
+	imageID, instanceType, sshKey, privateKeyPath string) (cloudprovider.Cloud, error) {
+	// File, dir, credential account sanity checks.
+	kubeconfigPath, err := cloudprovider.MakeValidAbsPath(kubeconfigPath)
+	if err != nil {
+		return nil, fmt.Errorf("error resolving path for kubeconfig file, %v", err)
+	}
+	credentialPath, err = cloudprovider.MakeValidAbsPath(credentialPath)
+	if err != nil {
+		return nil, fmt.Errorf("error resolving path for credentials file, %v", err)
+	}
+	resourceTrackerDir, err = cloudprovider.MakeValidAbsPath(resourceTrackerDir)
+	if err != nil {
+		return nil, fmt.Errorf("error resolving path for resource tracker directory, %v", err)
+	}
+
+	// Create a new client of the given OpenShift cluster based on service account created in test cluster.
+	oc, err := getOpenShiftClientForSA(kubeconfigPath)
```

Comment:
Is this the only method changing? If yes, can you just get the serviceaccount kubeconfig and use it at [this](https://github.com/openshift/windows-machine-config-bootstrapper/pull/178/files#diff-d69f962c88c5c87fe0fa39e99bde78b6R113) place?

Reply:
The serviceaccount config is in the form of rest.Config object and not a config file, which is used to get OpenShift client

Reply:
This makes me think you need to be passing the `oc` client into the factory and not the `kubeconfig`. That way you wouldn't need this function just for CI.